hydra-node/src/Hydra/Node.hs

```diff
@@ -223,15 +232,20 @@ processNextEvent ::
   IsChainState tx =>
   HydraNode tx m ->
   Event tx ->
+  Word64 ->
   STM m (Outcome tx)
-processNextEvent HydraNode{nodeState, ledger, env} e =
+processNextEvent HydraNode{nodeState, ledger, env} e nextStateChangeID = do
   modifyHeadState $ \s ->
     let outcome = computeOutcome s e
      in (outcome, aggregateState s outcome)
  where
   NodeState{modifyHeadState} = nodeState
 
-  computeOutcome = Logic.update env ledger
+  computeOutcome = Logic.update env ledger nextStateChangeID
+
+-- NOTE(Elaine): think you shouldnt be able to put all above into a single atomically call, the types shouldnt allow that
+-- some of the sinks won't be atomic, even if the disk-based persistence is
+-- and actually it doesn't matter if we accidentally do the same event twice because the PERSISTENCE is in charge of at-least-once semantics
```

Comment:
What exactly should not be in a single atomically call? In fact, I do not expect this function above to change at all?